homeassistant/components/myio/__init__.py

```diff
@@ -0,0 +1,109 @@
+"""The myIO integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from myio.comms_thread import CommsThread  # pylint: disable=import-error
+
+from homeassistant.const import CONF_NAME
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import slugify
+
+from .const import CONF_REFRESH_TIME, DOMAIN, SENSOR
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [SENSOR]
+
+COMMS_THREAD = CommsThread()
+
+
+async def async_setup(hass, config):
+    """Set up the myIO-server component."""
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up config entry."""
+    _server_name = slugify(config_entry.data[CONF_NAME])
+
+    hass.data.setdefault(DOMAIN, {})[_server_name] = {}
+
+    # hass.data[DOMAIN][_server_name]["state"] = "Offline"
+    hass.states.async_set(f"{_server_name}.state", "Offline")
+
+    _refresh_timer = config_entry.options.get(CONF_REFRESH_TIME, 4)
+
+    def server_data():
+        """Return the server data dictionary database."""
+        return hass.data[DOMAIN][_server_name]
+
+    async def setup_config_entries():
+        """Set up myIO platforms with config entry."""
+
+        for component in PLATFORMS:
+            hass.async_create_task(
+                hass.config_entries.async_forward_entry_setup(config_entry, component)
+            )
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        was_offline = False
+
+        _temp_server_state = hass.states.get(f"{_server_name}.state").state
+
+        if _temp_server_state == "Offline":
+            hass.states.async_set(f"{_server_name}.available", False)
+            was_offline = True
+
+        # Pull fresh data from server.
+        [hass.data[DOMAIN][_server_name], _temp_server_state] = await COMMS_THREAD.send(
+            server_data=server_data(),
+            server_status=_temp_server_state,
+            config_entry=config_entry,
+            _post=None,
+        )
+
+        hass.states.async_set(f"{_server_name}.state", _temp_server_state)
```

Comment:
use a regular entity if it needs to be in the state machine, otherwise store in hass.data